ownership-chain/pallets/laos-evolution/src/lib.rs

```diff
@@ -194,6 +211,36 @@ impl<T: Config> EvolutionCollection<AccountIdOf<T>, TokenUriOf<T>> for Pallet<T>
 
 		Ok(())
 	}
+
+	fn enable_public_minting(who: AccountIdOf<T>, collection_id: CollectionId) -> DispatchResult {
+		ensure!(
+			CollectionOwner::<T>::contains_key(collection_id),
+			Error::<T>::CollectionDoesNotExist
+		);
+		ensure!(CollectionOwner::<T>::get(collection_id) == Some(who), Error::<T>::NoPermission);
+		CollectionPublicMintingEnabled::<T>::insert(collection_id, ());
+		Self::deposit_event(Event::PublicMintingEnabled { collection_id });
+		Ok(())
+	}
+
+	fn disable_public_minting(who: AccountIdOf<T>, collection_id: CollectionId) -> DispatchResult {
+		ensure!(
+			CollectionOwner::<T>::contains_key(collection_id),
+			Error::<T>::CollectionDoesNotExist
+		);
+		ensure!(CollectionOwner::<T>::get(collection_id) == Some(who), Error::<T>::NoPermission);
+		CollectionPublicMintingEnabled::<T>::remove(collection_id);
+		Self::deposit_event(Event::PublicMintingDisabled { collection_id });
+		Ok(())
+	}
+
+	fn is_public_minting_enabled(collection_id: CollectionId) -> Result<bool, DispatchError> {
+		ensure!(
```

Comment:
this should not be part of this function. It should basically tell if public minting is enabled or not. check for collection existence should be somewhere else. It also looks strange that a boolean function returns `Result`

Reply:
yeah good point